crates/bevy_window/src/window.rs

```diff
@@ -929,124 +869,52 @@ pub enum MonitorSelection {
     Index(usize),
 }
 
-/// Describes the information needed for creating a window.
-///
-/// This should be set up before adding the [`WindowPlugin`](crate::WindowPlugin).
-/// Most of these settings can also later be configured through the [`Window`](crate::Window) resource.
+/// Marker component for the window considered the primary window.
 ///
-/// See [`examples/window/window_settings.rs`] for usage.
-///
-/// [`examples/window/window_settings.rs`]: https://github.com/bevyengine/bevy/blob/latest/examples/window/window_settings.rs
-#[derive(Debug, Clone, PartialEq)]
-#[cfg_attr(feature = "serialize", derive(serde::Serialize, serde::Deserialize))]
-pub struct WindowDescriptor {
-    /// The requested logical width of the window's client area.
-    ///
-    /// May vary from the physical width due to different pixel density on different monitors.
-    pub width: f32,
-    /// The requested logical height of the window's client area.
-    ///
-    /// May vary from the physical height due to different pixel density on different monitors.
-    pub height: f32,
-    /// The position on the screen that the window will be placed at.
-    ///
-    /// The monitor to place the window on can be selected with the `monitor` field.
-    ///
-    /// Ignored if `mode` is set to something other than [`WindowMode::Windowed`]
-    ///
-    /// `WindowPosition::Automatic` will be overridden with `WindowPosition::At(Vec2::ZERO)` if a specific monitor is selected.
-    pub position: WindowPosition,
-    /// The monitor to place the window on.
-    pub monitor: MonitorSelection,
-    /// Sets minimum and maximum resize limits.
-    pub resize_constraints: WindowResizeConstraints,
-    /// Overrides the window's ratio of physical pixels to logical pixels.
-    ///
-    /// If there are some scaling problems on X11 try to set this option to `Some(1.0)`.
-    pub scale_factor_override: Option<f64>,
-    /// Sets the title that displays on the window top bar, on the system task bar and other OS specific places.
+/// Currently this is assumed to only exist on 1 entity at a time.
+#[derive(Default, Debug, Component, PartialEq, Eq, PartialOrd, Ord, Copy, Clone, Reflect)]
+#[reflect(Component)]
+pub struct PrimaryWindow;
```

Comment:
Are we planning on using PrimaryWindow as a resource or as a marker component atm? 

Suggest we either remove this, or change all code to use marker component instead of resource. Having both I suspect is confusing?

Reply:
I refactored it to be used as a marker component, there shouldn't be any parts using it as a resource anymore